### PR TITLE
Rmd report template

### PR DESCRIPTION
**Pull request by ahasverus**
This PR aims to generate a `Rmd` report. It's link to #35 and maybe #34.

Development workflow:
- [x] Create template draft
- [x] Create new `Rmd` from template
- [x] Replace default values in file (title, author, dataset names)
- [x] Pass users data in `Rmd`

Note: the final template will be written later. This PR can be closed before.

### CODE REVIEW

File: inst/templates/template_report.Rmd [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(
+  fig.width  = 7,
+  fig.height = 7,
+  dpi        = 100,
+  out.width  = "100%",
+  echo       = FALSE,
+  eval       = TRUE
+)
+```
```
Comment (Rekyt) [inline inst/templates/template_report.Rmd:40]: We have to call the package to work :)

````suggestion
)

library("funbiogeo")
```
````

File: inst/templates/template_report.Rmd [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+## Sites x locations data
+
+The sites x locations data consists in `r nrow(site_locations)` sites.
+
+
+\
+
+
+```{r site-locations}
+print(site_locations)
```
Comment (Rekyt) [inline inst/templates/template_report.Rmd:117]: To avoid showing too many lines

```suggestion
head(site_locations)
```

File: R/fb_make_report.R [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  
+  
+  # Copy template ----
+  
+  invisible(
+    file.copy(system.file(file.path("templates", "template_report.Rmd"), 
+                          package = "funbiogeo"), path, overwrite = TRUE
+  ))
+  
+
```
Comment (Rekyt) [inline R/fb_make_report.R:128]:
```suggestion
  message("The file ", path, "was created!\nOpen it then use knitr::knit() or rmarkdown::render() to render it.")
```

File: R/fb_make_report.R [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    xfun::gsub_file(path, "\"{{author}}\"", paste0("\"", author, "\""), 
+                    fixed = TRUE)
+  }
+  
+  xfun::gsub_file(path, "\"{{title}}\"", paste0("\"", title, "\""), 
+                  fixed = TRUE)
+  
+  
+  # Replace data names ----
+
```
Comment (Rekyt) [inline R/fb_make_report.R:151]:
```suggestion
  # Replace data names ----
  
  message("Specified object names\n",
                  "species x traits: \"", species_traits_name, "\"\n",
                  "site x species: \"", site_species_name, "\"\n",
                  "site x locations: \"", site_locations_name"\"")
             
```

**Review by Rekyt — COMMENTED** [review]
Congrats for this PR! It's a great proof-of-concept!

I still have some suggestions:

- We should modify the template to add more plots and text to explain what's in the data (I can do this :))
- The first section of the report should be about formatting the primary objects or explain it (maybe a section with REPLACE ME part or something like this).
- Maybe we can open the report interactively using a similar mechanism as `usethis::use_r()`. This would add a `Suggests` on the `rstudioapi` package but would be tremendously useful and explicit.

**Comment by ahasverus** [comment]
Thanks for you feedback!
I will implement the code to open interactively the file, I have done something similar for `rcompendium`

**Pull request merged**